grails-app/controllers/org/pih/warehouse/product/AttributeController.groovy

```diff
@@ -23,7 +25,10 @@ class AttributeController {
 
     def list() {
         params.max = Math.min(params.max ? params.int('max') : 10, 100)
-        [attributeInstanceList: Attribute.list(params), attributeInstanceTotal: Attribute.count()]
+        params.offset = params.int('offset') ?: 0
```

Comment:
We have been using a ternary operator for assigning default values for the params across the whole application so this isn't a change request but rather a suggestion to use the built-in default value parameter for these cases.
So instead of 
```groovy
params.max = Math.min(params.max ? params.int('max') : 10, 100)
params.offset = params.int('offset') ?: 0
```
we can do
```groovy
params.max = Math.min(params.int('max', 10), 100)
params.offset = params.int('offset', 0)
```
And if we have cases where we need to assign a default value for a string, unfortunately, there is no method `params.string('my_string_param', 'default string value')`
but rather
`params.getOrDefault('my_string_param', 'default string value')`

I am making this comment just to point out that this is available and this way makes the code a bit more sophisticated although it is an insignificant change.

Reply:
params.getOrDefault() is ugly af but I'm ok with using it. Odd that there isn't an equivalent for strings. I'll probably continue doing string parameters like JSB recommends in the following response.

https://stackoverflow.com/a/24266854/136597